internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -58,10 +59,10 @@ trait WithPattern { protected def Pattern: Regex }
 import java.lang.{ Long => BoxedLong }
 
 /** Define the hash of the file contents. It's a typical stamp for compilation sources. */
-final class Hash(val value: Array[Byte]) extends StampBase {
-  val hexHash = IOHash.toHex(value)
+final class Hash private[inc] (val hexHash: String) extends StampBase {
```

Comment:
No, I don't think that's possible. `Hash` is instantiated from two different objects: `Stamp` and `Stamper`, and wrapping them into Hash's companion object is not a valid solution.